SportsHub/SportsHub.Api/Controllers/CommentController.cs

```diff
@@ -1,33 +1,99 @@
+        private readonly ICommentService _commentService;
         private readonly IMapper _mapper;
+        private readonly IValidator<InputCommentDTO> _commentValidator;
+        private readonly IGenerateModelStateDictionary _generateModelStateDictionary;
 
-        public CommentController(ICommentService service, IMapper mapper)
+        public CommentController(ICommentService service, IMapper mapper, IValidator<InputCommentDTO> commentValidator, IGenerateModelStateDictionary generateModelStateDictionary)
         {
-            _service = service;
+            _commentService = service;
             _mapper = mapper;
+            _commentValidator = commentValidator;
+            _generateModelStateDictionary = generateModelStateDictionary;
         }
 
         [HttpGet("GetByArticle")]
-        public async Task<IActionResult> GetByArticleAsync(int articleId)
+        public async Task<ActionResult<IEnumerable<CreateCommentRequest>>> GetByArticleAsync(int articleId)
         {
-            var comments = await _service.GetByArticleAsync(articleId);
+            var comments = await _commentService.GetByArticleAsync(articleId);
 
             if (!comments.Any())
             {
-                return BadRequest($"No comments for this article");
+                return Ok(ValidationMessages.NoCommentsForArticle);
             }
 
-            return Ok(comments);
+            var commentsResponse = _mapper.Map<List<CreateCommentRequest>>(comments);
+
+            return Ok(commentsResponse);
+        }
+
+        [Authorize]
+        [HttpPost("PostComment")]
+        public async Task<ActionResult> PostCommentAsync([FromBody] InputCommentDTO commentInput)
+        {
+            var validationResult = await _commentValidator.ValidateAsync(commentInput);
+
+            if (!validationResult.IsValid)
+            {
+                var response = _generateModelStateDictionary.modelStateDictionary(validationResult);
+                return ValidationProblem(response);
+            }
+
+            var comment = _mapper.Map<CreateCommentDTO>(commentInput);
+
+            var created = await _commentService.AddCommentAsync(comment);
+
+            if (!created)
+            {
+                return BadRequest(ValidationMessages.UnableToAddComment);
+            }
+
+            return Ok(ValidationMessages.CommentAddedSuccessfully);
+        }
+
+        [Authorize]
+        [HttpPost("LikeComment")]
+        public async Task<ActionResult> LikeCommentAsync(int commentId)
+        {
+            var result = await _commentService.LikeCommentAsync(commentId);
+
+            if (!result)
+            {
+                return BadRequest(ValidationMessages.NoSuchComment);
+            }
+
+            return Ok(ValidationMessages.CommentSuccessfullyLiked);
+        }
+
+        [Authorize]
+        [HttpPost("DislikeComment")]
+        public async Task<ActionResult> DislikeCommentAsync(int commentId)
+        {
```

Comment:
maybe there should be one method enough? and in parameter can be something like - isLike or LikeType ("like" or dislike)

Reply:
@Banovvv will you change this?

Reply:
Sorry for the late response, can it be left as it is?

Reply:
Create a tech debt ticket and describe their problem, because the approach is not correct. It looks like one user can like some posts hundreds of times.